tests/layer_tests/tensorflow_tests/test_tf_SegmentSum.py

```diff
@@ -69,3 +69,49 @@ def test_segment_sum_different_types(self, params, ie_device, precision, ir_vers
         self._test(*self.create_segment_sum_net(**params),
                    ie_device, precision, ir_version, temp_dir=temp_dir,
                    use_legacy_frontend=use_legacy_frontend)
+
+class TestSegmentSumComplex(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
+        rng = np.random.default_rng()
+        assert "data:0" in inputs_info
+        assert "segment_ids:0" in inputs_info
+        data_shape = inputs_info["data:0"]
+        segment_ids_shape = inputs_info["segment_ids:0"]
+        inputs_data = {}
+        inputs_data["data:0"] = rng.random(data_shape).astype(np.float32)
+        inputs_data["segment_ids:0"] = rng.integers(
+            low=0, high=data_shape[0], size=segment_ids_shape
+        ).astype(np.int32)
+        return inputs_data
+
+    def create_segment_sum_net(self, data_shape, segment_ids_shape):
+        tf.compat.v1.reset_default_graph()
+        # Create the graph and model
+        with tf.compat.v1.Session() as sess:
+            data = tf.compat.v1.placeholder(np.float32, data_shape, "data")
```

Comment:
For `data` input to `SegmentSum`. please create two input placeholders for `real` and `imag` parts and pass them to `Complex` operation. The `Complex` output will be `data` for `SegmentSum`. As example, see here: https://github.com/openvinotoolkit/openvino/blob/master/tests/layer_tests/tensorflow_tests/test_tf_Rank.py#L54

Reply:
This is needed, because OV supports complex type only for intermediate layers (via floating-point tensor representation). It does not support for model inputs and outputs of complex type because there is no native complex type in OV.